tools/syz-declextract/README.md

```diff
@@ -0,0 +1,41 @@
+# syz-declextract
+## Linux Kernel (For testing purposes)
+```
+export KERNEL=$PWD/linux-stable
+git clone \
+https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git $KERNEL
+cd $KERNEL
+make CC=clang defconfig # Having clang as the compiler is optional but removes
+                        # erros later on
+./scripts/config -e FTRACE_SYSCALLS
+make CC=clang olddefconfig
+make CC=clang -j`nproc` # kernel has to be built at least once for the script
+                        # to work
+./scripts/clang/gen_compile_commands.py
+```
+## LLVM Project
+```
+LLVM=$PWD/llvm-project
+git clone https://github.com/llvm/llvm-project.git $LLVM
+echo 'add_clang_executable(syz-declextract syz-declextract/syz-declextract.cpp)
+target_link_libraries(syz-declextract PRIVATE clangTooling)' >> \
+$LLVM/clang/CMakeLists.txt
+```
+## syz-declextract
+```
+mkdir $LLVM/clang/syz-declextract
+```
+Download `syz-declextract.cpp` file and add it to `$LLVM/clang/syz-declextract`
+directory
+```
+SYZ=$PWD/syz
+mkdir $SYZ && cd $SYZ
+cmake -DLLVM_ENABLE_PROJECTS="clang" -DCMAKE_BUILD_TYPE=Release \
+-DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_C_COMPILER=clang \
+-DCMAKE_CXX_COMPILER=`which clang++` $LLVM/llvm/
+make CC=clang -j`nproc` syz-declextract
+```
+## Example
+```
+./bin/syz-declextract $KERNEL/fs/read_write.c | less # or any other .c file
```

Comment:
It does not produce warnings, how does it find compile_commands.json? Does it look up the directory tree to find $KERNEL/compile_commands.json?